### PR TITLE
Make i18n work with node/browserify

### PR DESCRIPTION
**Pull request by simon04**
The locale/language is taken from `optional_conf_parm`. No direct
interaction with `i18next` module is required.

### CODE REVIEW

**Comment by simon04** [comment]
Have you already found time to consider this pull request?

File: opening_hours.js [hunk 1]
```diff
@@ -4251,7 +4261,7 @@
 					user_conf[key] = default_prettify_conf[key];
 			}
 
-			if (typeof user_conf['locale'] === 'string' && user_conf['locale'] !== 'en') {
+			if (typeof moment !== 'undefined' && typeof user_conf['locale'] === 'string' && user_conf['locale'] !== 'en') {
```
Comment (ypid) [inline opening_hours.js]: Please be more specific. E.g. use `typeof moment === 'object'`.

**Comment by simon04** [comment]
Changed to `typeof moment === 'object'` …

**Comment by ypid** [comment]
Looks good. I just need to make the evaluation tool and the map work again.

You updated the test output. The interesting thing is that when I checkout 72e5f091d7094e8d878af6a4c8a280c74069b8b2 I get the old version. I guess this is platform/version depended. What version of NodeJS are you using and what Distro?

File: locales/core.js [hunk 2]
```diff
@@ -1,3 +1,9 @@
+var i18n = i18n;
```
Comment (ypid) [inline locales/core.js]: I don’t quite get that. Is this needed for Nashorn/Java?
For the browser and NodeJS it works without it.

File: locales/core.js [hunk 3]
```diff
@@ -1,3 +1,9 @@
...
+if (require && module) {
+  i18n = require('i18next-client');
```
Comment (ypid) [inline locales/core.js]: For my i18n also works as `var i18n`.

**Comment by ypid** [comment]
Well done. Thanks very much …

**Comment by simon04** [comment]
```
$ iojs --version
v3.0.0
$ npm --version
2.13.3
```
… using Arch Linux and https://aur.archlinux.org/packages/iojs-bin/


But I get the same output with
```
$ node --version
v0.12.7
$ npm --version
2.13.5
```
using NodeJs from the official repository https://www.archlinux.org/packages/community/x86_64/nodejs/

**Comment by simon04** [comment]
Thank you for your review. I simplified it to `var i18n = require('i18next-client');`

**Comment by ypid** [comment]
Thanks again and I am looking forward to see the error/warning translation feature in JOSM :+1: :smile:

**Pull request merged**